src/match/controllers/update_match_controller.py

```diff
@@ -0,0 +1,144 @@
+import json
+
+from api.utils import failure_response
+from api.utils import success_response
+from django.core.exceptions import ValidationError
+from location.models import Location
+from match import match_status
+from match.models import Match
+from match.validators import validate_int_list
+from match.validators import validate_times_list
+from rest_framework import status
+
+
+class UpdateMatchController:
+    def __init__(self, match_id, request, data, serializer):
+        self._match_id = match_id
+        self._request = request
+        self._serializer = serializer
+        self._data = data
+
+    def process(self):
+        """Update match fields that have changed."""
+        self._match = Match.objects.filter(id=self._match_id)
+        if not self._match:
+            return failure_response("Match does not exist.")
+        self._match = self._match[0]
+        status = self._data.get("status")
+        accepted_ids = self._data.get("accepted_ids")
+        proposed_meeting_times = self._data.get("proposed_meeting_times")
+        proposed_locations = self._data.get("proposed_locations")
+        meeting_location = self._data.get("meeting_location")
+        meeting_time = self._data.get("meeting_time")
+        output_messages = []
+        output_messages.append(
+            self._validate_acceptors_and_times(accepted_ids, proposed_meeting_times)
+        )
+        output_messages.append(self._validate_status(status))
+        self._update_proposer_and_acceptors(status)
+        self._modify_attribute("status", status)
+        self._modify_attribute("accepted_ids", accepted_ids)
+        self._modify_attribute("proposed_meeting_times", proposed_meeting_times)
+        self._modify_attribute("meeting_time", meeting_time)
+        if proposed_locations is not None:
+            output_messages.append(self._update_proposed_locations(proposed_locations))
+        if meeting_location is not None:
+            output_messages.append(self._update_meeting_location(meeting_location))
+        self._validate_match_status()
+        for error in output_messages:
+            if error is not None:
+                return error
+        self._match.save()
+        return success_response()
```

Comment:
if this is just returning a `success_response` you should make this into an asynchronous task -- you should make an issue for this and make another PR for it tho